R/try.R

```diff
@@ -13,24 +13,24 @@
 #' FunDoesNotExist(1:10) %except% MEAN(1:10) %except% mean(1:10)
 #' FunDoesNotExist(1:10) %except% (MEAN(1:10) %except% mean(1:10))
 `%except%` <- function(try, except) {
-
-    call <- sys.call(-1)
-    env <- parent.frame()
-    try <- substitute(try)
-    fallback <- substitute(except)
+    # Need to capture these in the evaluation frame of `%except%` but only want
+    # to do the work if there's an error
+    delayedAssign("call", sys.call(-1))
```

Comment:
I think this is a nice technique to evaluate all the code in the correct frame, but only evaluate it if it's actually needed.